conans/client/rest/remote_credentials.py

```diff
@@ -58,12 +58,12 @@ def auth(self, remote, user=None):
         if env_passwd is not None:
             if env_user is None:
                 raise ConanException("Found password in env-var, but not defined user")
-            return env_user, env_passwd
+            return env_user, env_passwd, False
 
         # If not found, then interactive prompt
         ui = UserInput(self._global_conf.get("core:non_interactive", check_type=bool))
         input_user, input_password = ui.request_login(remote.name, user)
-        return input_user, input_password
+        return input_user, input_password, True
```

Comment:
Seems like `UserInput` throws if `core:non_interactive` is True, so at this point we're always interactive 👍